distribution/scripting.md

```diff
@@ -45,16 +34,74 @@ registerPlugin({
+
+**Local plugins** will load on any client in multiplayer, which will allow each player to bring any local script for their own game without other players in a multiplayer server also needing to also enable the same script.
+
+Local plugins **cannot** alter the game state directly, but they **can** interact with the game by mimicking a player's actions through the built-in “game actions”. Local plugins may also use custom actions, but clients that do not have the plugin will not execute the registered callbacks, which may lead to a desync.
+
+**Remote plugins** will only load in multiplayer when installed on the server. When players join a server, the remote script will be downloaded to the client and will stay active for the duration of the multiplayer session. Any remote plugins installed on the client will not load in multiplayer.
+
+Remote plugins **can** interact with the game by mimicking a player's actions through the built-in “game actions”. Remote scripts **can** also alter the game state directly, but only within the “execute” context of custom game actions. With custom actions, OpenRCT2 can ensure that the game state mutations are executed synchronously across all connected clients.
+
+**Intransient plugins** in multiplayer have the same restrictions as local plugins. They will stay loaded and only interact with the game through game actions.
+
+#### Example usage
+
+- Local plugins tend to provide extra tools for productivity with the built-in game actions, or new windows containing information about aspects of the game or the park.
+- Remote plugins can be tools that have to mutate game state more directly, or multiplayer server functionality like welcome messages, anti-spam, or kick/ban tools.
+- Intransient plugins could provide functionality across different screens such as the title screen, and can be used for plugins like the title sequence editor or global dashboards.
+
+### Plugin licence
+
+The authors must also define a licence for the plugin, making it clear to the community whether that plugin can be altered, copied, etc. A good reference material is listed on [ChooseALlicense](https://choosealicense.com/appendix/), try to pick one of them and use its corresponding identifier, as listed on [SPDX](https://spdx.org/licenses/).
+
+### Hot reload
+
+The hot reload feature can be enabled by editing your `config.ini` file and setting `enable_hot_reloading` to `true` under `[plugin]`. When this is enabled, the game will auto-reload the script in real-time whenever you save your JavaScript file. This allows rapid development of plugins as you can write code and quickly preview your changes, such as closing and opening a specific custom window on startup. A demonstration of this can be found on YouTube: [OpenRCT2 plugin hot-reload demo](https://www.youtube.com/watch?v=jmjWzEhmDjk)
+
+## Game actions
+
+Game actions are actions that players can invoke in games. Here is an example flow of a game action such as opening the park:
```

Comment:
Please add some intuition how to use actions (especially compared to direct data access):
- for everything that the player can do to change the game state, there is a game action
- plugins can execute any of these actions as if they were a player (it is actually never mentioned here)
- **using actions is the prefered way to change the game state**, since it automatically ensures multiplayer-safety and sanity checks (cash, clearance, ...)
- but, if the plugin really needs to use direct game state changes, it is strongly advised to make the plugin remote and wrap all game state changes in custom actions. Otherwise, the plugin will not work in multiplayer and likely cause disconnects. Even if you as the plugin developer do not want to specifically target multiplayer, it is still better to follow these guidelines to prevent disconnects of your users.

Please copy or modify as you see fit.
As a side note, if it wouldn't absolutely destroy backwards compatibility, I would really like to disable all direct game state changes outside of custom actions.